src/main/kotlin/snyk/advisor/SnykAdvisorModel.kt

```diff
@@ -0,0 +1,98 @@
+
+    private lateinit var dropCacheRunnable: () -> Unit
+
+    init {
+        dropCacheRunnable = {
+            name2Info.clear()
+            alarmToDropCache.addRequest(dropCacheRunnable, H24_IN_MILLISECONDS)
+        }
+        dropCacheRunnable()
+    }
+
+    fun getScore(project: Project?, packageManager: AdvisorPackageManager, packageName: String): Int? {
+
+        name2Info[packageName]?.let { info ->
+            return (info.score * 100).toInt()
+        }
+
+        /** Ignore requests for that [packageName] during [IGNORE_REQUESTS_DELAY] after first ever request
+         * to avoid create server requests while __user is typing__ */
+        val firstRequest = packages2FirstRequest.put(packageName, Instant.now()) ?: return null
+        if (firstRequest.plusMillis(IGNORE_REQUESTS_DELAY) > Instant.now()) return null
+
+        /** Proceed requests in batch, by waiting for [SCORE_REQUESTS_BATCHING_DELAY] after last request */
+        if (packagesRequestDelayed.add(packageName)) {
+            alarmToRequestScore.cancelAllRequests()
+            alarmToRequestScore.addRequest({
+                // exclude packages received already while we been waiting/delaying
+                val packagesToRequest = packagesRequestDelayed - packagesRequested - name2Info.keys
+                if (packagesRequested.addAll(packagesToRequest)) {
+                    log.debug("Requested: $packagesToRequest")
+                    advisorService.requestPackageInfos(
+                        project = project,
+                        packageManager = packageManager,
+                        packageNames = packagesToRequest.toList(),
+                        onPackageInfoReady = { name, packageInfo ->
+                            packagesRequested.remove(name)
+                            name2Info[name] = packageInfo
+                            log.debug("Advisor info received for $name : $packageInfo")
+                        },
+                        onFailGetInfo = { name ->
+                            packagesRequested.remove(name)
+                            log.debug("Failed to get info for $name")
+                        }
+                    )
+                }
+                packagesRequestDelayed.clear()
+                packages2FirstRequest.clear()
+            }, SCORE_REQUESTS_BATCHING_DELAY)
+        }
+        return null
+    }
+
+    // todo(pavel): internal ?
+    @TestOnly
+    fun setAdvisorService(mockedAdvisorService: AdvisorService) {
+        this.advisorService = mockedAdvisorService
+    }
+
+    @TestOnly
+    fun getPackagesRequestDelayed() = packagesRequestDelayed
+
+    @TestOnly
+    fun getPackagesRequested() = packagesRequested
+
+    override fun dispose() {}
+
+    companion object {
+        private const val H24_IN_MILLISECONDS = 24 * 60 * 60 * 1000
+        const val IGNORE_REQUESTS_DELAY: Long = 1000 //ms
+        const val SCORE_REQUESTS_BATCHING_DELAY = 1000 //ms
```

Comment:
How about adding a suffix to show that it's milliseconds? It will be simpler to read.